.github/workflows/windows_cmake.yml

```diff
@@ -0,0 +1,163 @@
+name: Windows CMake build
+
+on:
+  push:
+    branches: [ RC_1_2 ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ RC_1_2 ]
+
+env:
+  # openssl: 1.1.1g#1
+  # boost: 1.73.0
+  VCPKG_COMMIT: 27a2418e91179d8607862348d7b498558e98a0ab
+  VCPKG_DEST_WIN: C:\libt_tools\vcpkg
+  VCPKG_TRIPLET: "x64-windows"
+  BUILD_VARIANT: "shared"
+  LIBT_STATIC_RT: "OFF"
+  LIBT_BSL: "ON"
+  LIBT_TESTS: "ON"
+  LIBT_DEPR_FUN: "ON"
+  LIBT_PY_BINDINGS: "ON"
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+
+  ci_windows_cmake:
+    name: Windows CMake build (w/ vcpkg)
+
+    strategy:
+      matrix:
+        build_config: [Release, Debug]
+        build_variant: [shared, static]
+        deprecated_functions: [yes_depr_fun, no_depr_fun]
+        build_tests: [yes_tests, no_tests]
+        python_bindings: [yes_py_bindings, no_py_bindings]
+        exclude:
+          # python bindings require building with shared libs
+          - build_variant: static
+            python_bindings: yes_py_bindings
+          # FIXME: non-static build with python bindings is failing, presumably due to a boost port bug in vcpkg
+          # See https://github.com/microsoft/vcpkg/issues/5097
+          - build_variant: shared
+            build_config: Debug
+            python_bindings: yes_py_bindings
+          # tests require deprecated functions
```

Comment:
this passes for me in `tests`, on macOS:

```
$ b2 deprecated-functions=off link=static -j8 -l250 cxxstd=11
```

Reply:
> \# tests require deprecated functions

This might be due to some comment and read (and possibly misinterpreted) long ago. Will look into it.

Reply:
@arvidn 

> > # tests require deprecated functions
> 
> This might be due to some comment I read (and possibly misinterpreted) long ago. Will look into it.

Actually, it's right here:
https://github.com/arvidn/libtorrent/blob/8ee5c84e6434cdb8416ce68a6202e50dda7743bc/CMakeLists.txt#L631

Reply:
fixed by rebasing to >= 74a7dd9c66fe6818cdfeb9ad8bfc292088924443